blockdevice/util/disks.go

```diff
@@ -107,3 +108,34 @@ func GetDisks() ([]*Disk, error) {
 
 	return disks, nil
 }
+
+var (
+	pathRegexp = regexp.MustCompile(`ID_PATH=(.*)`)
+	uuidRegexp = regexp.MustCompile(`ID_PART_TABLE_UUID=(.*)`)
+)
+
+// DevicePath gets device path using disk dev path or disk name and /sys/block data.
+func DevicePath(dev string) (string, error) {
+	return runUdevadm("path_id", dev, pathRegexp)
+}
+
+// DeviceUUID gets device uuid using disk dev path or disk name and /sys/block data.
+func DeviceUUID(dev string) (string, error) {
+	return runUdevadm("blkid", dev, uuidRegexp)
+}
+
+func runUdevadm(subcommand, dev string, re *regexp.Regexp) (string, error) {
+	sysblock := filepath.Join("/sys/block", filepath.Base(dev))
+
+	out, err := cmd.Run("udevadm", "test-builtin", subcommand, sysblock)
```

Comment:
any chance we would do it without `udevadm` ? is it too much code?

Reply:
yeah, looked like too much code and bit of reverse engineering of how they generate it.
But if we decide to avoid it at all costs I think it's doable.